docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,470 @@
+  (`#270 <https://github.com/oneapi-src/oneMKL/issues/270>`_).
+
+
+.. _build_invocation_examples_dpcpp:
+
+CMake invocation examples
+##########################
+
+Build oneMKL with support for x86 CPU, Intel GPU, and Nvidia GPUs with tests
+disabled using the Ninja build system:
+
+.. code-block:: bash
+
+  cmake $ONEMKL_DIR \
+      -GNinja \
+      -DCMAKE_CXX_COMPILER=clang++ \
+      -DCMAKE_C_COMPILER=clang \
+      -DENABLE_MKLGPU_BACKEND=False \
+      -DENABLE_MKLCPU_BACKEND=False \
+      -DENABLE_CUFFT_BACKEND=True \
+      -DENABLE_CUBLAS_BACKEND=True \
+      -DENABLE_CUSOLVER_BACKEND=True \
+      -DENABLE_CURAND_BACKEND=True \
+      -DBUILD_FUNCTIONAL_TESTS=False
+
+``$ONEMKL_DIR`` points at the oneMKL source directly. The x86 CPU (``MKLCPU``)
+and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled
+here. The backends for Nvidia GPUs must all be explicilty enabled. The tests are
+disabled, but the examples will still be built.
+
+Building oneMKL with support x86 CPU, Intel GPU, and AMD GPUs with tests
+disabled:
+
+.. code-block:: bash
+
+  cmake $ONEMKL_DIR \
+      -DCMAKE_CXX_COMPILER=clang++ \ 
+      -DCMAKE_C_COMPILER=clang \
+      -DENABLE_MKLCPU_BACKEND=False \ 
+      -DENABLE_MKLGPU_BACKEND=False \
+      -DENABLE_ROCFFT_BACKEND=True  \ 
+      -DENABLE_ROCBLAS_BACKEND=True \
+      -DENABLE_ROCSOLVER_BACKEND=True \ 
+      -DHIP_TARGETS=gfx90a \
+      -DBUILD_FUNCTIONAL_TESTS=False
+
+``$ONEMKL_DIR`` points at the oneMKL source directly. The x86 CPU (``MKLCPU``)
+and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled
+here. The backends for AMD GPUs must all be explicilty enabled. The tests are
+disabled, but the examples will still be built.
+
+
+Build oneMKL for the DFT domain only with support for x86 CPU, Intel GPU, AMD
+GPU and Nvidia GPU with testing enabled:
+
+.. code-block:: bash
+
+  cmake $ONEMKL_DIR \ 
+      -DCMAKE_CXX_COMPILER=icpx \
+      -DCMAKE_C_COMPILER=icx \ 
+      -DENABLE_ROCFFT_BACKEND=True \
+      -DENABLE_CUFFT_BACKEND=True \
+      -DTARGET_DOMAINS=dft \
+      -DBUILD_EXAMPLES=False
+
+Note that this is not a supported configuration, and requires Codeplay's oneAPI
+for `AMD <https://developer.codeplay.com/products/oneapi/amd/home/>`_ and
+`Nvidia <https://developer.codeplay.com/products/oneapi/nvidia/home/>`_ GPU
+plugins. Like the above example, the MKLCPU and MKLGPU backends are enabled by
+default, with backends for Nvidia GPU and AMD GPU explicitly enabled.
+``-DTARGET_DOMAINS=dft`` causes only DFT backends to be built. If this was not
+set, the backend libraries to enable the use of BLAS, LAPACK and RNG with MKLGPU
+and MKLCPU would also be enabled. The build of examples is disabled. Since
+functional testing was not disabled, tests would be built.
+
+.. _project_cleanup:
+
+Project Cleanup
+###############
+
+Most use-cases involve building the project without the need to cleanup the
```

Comment:
```suggestion
Most use-cases involve building the project without the need to clean up the
```

Reply:
Done in [cbd83b4](https://github.com/oneapi-src/oneMKL/pull/510/commits/cbd83b473aa2c9c53c42fd21c95f9279ba16686b) and [ef02116](https://github.com/oneapi-src/oneMKL/pull/510/commits/ef021162bfa1dd44b8746c8b214cf248b16cbdf9)